metals/src/main/scala/scala/meta/internal/metals/BuildTargets.scala

```diff
@@ -229,6 +233,46 @@ final class BuildTargets() {
     }
   }
 
+  def jvmRunEnvironment(targetId: String): Future[JvmRunEnvironmentResult] =
+    jvmEnvironment(
+      targetId,
+      conn =>
+        conn.jvmRunEnvironment(
+          new JvmRunEnvironmentParams(
+            List(new BuildTargetIdentifier(targetId)).asJava
+          )
+        ),
+    )
+
+  def jvmTestEnvironment(targetId: String): Future[JvmTestEnvironmentResult] =
+    jvmEnvironment(
+      targetId,
+      conn =>
+        conn.jvmTestEnvironment(
+          new JvmTestEnvironmentParams(
+            List(new BuildTargetIdentifier(targetId)).asJava
+          )
+        ),
+    )
+
+  private def jvmEnvironment[T](
+      targetUri: String,
+      env: BuildServerConnection => Future[T],
+  ): Future[T] = {
+    // TODO if single
+    val results = for {
+      // (targetId, target) <- data.fromIterators(_.buildTargetInfo.iterator)
+      // if target.getDisplayName() == targetName
+      (connTargetId, conn) <- data.fromIterators(_.targetToConnection.iterator)
+      if connTargetId.getUri == targetUri
+    } yield env(conn)
```

Comment:
Is this really possible that one `targetUri` will match several connections?

Reply:
It shouldn't be possible, but if it is we might want to just present the first one.

Reply:
Changed the approach and now we are saving the env data.